src/nodemetadata.cpp

```diff
@@ -235,7 +237,8 @@ void NodeMetadataList::remove(v3s16 p)
 void NodeMetadataList::set(v3s16 p, NodeMetadata *d)
 {
 	remove(p);
-	m_data.insert(std::make_pair(p, d));
+	if (!d->empty())
+		m_data.emplace(p, d);
```

Comment:
Oops yes, this is entirely broken.
At the point new meta is inserted it is always empty, so nothing works.

A better approach to this would be to remove the meta if it is empty when a `NodeMetaRef` is garbage-collected (or maybe in `NodeMetaRef::handleFromTable` only?).